test-data/unit/check-classes.test

```diff
@@ -6596,3 +6596,45 @@ reveal_type(D() + "str")  # N: Revealed type is 'Any'
 
 reveal_type(0.5 + D1())  # N: Revealed type is 'Any'
 reveal_type(D1() + 0.5)  # N: Revealed type is '__main__.D1'
+
+[case testRefMethodWithDecorator]
+from typing import Type
+
+class A:
+    pass
+
+class B:
+    @staticmethod
+    def A() -> Type[A]:
+        return A
```

Comment:
Please put `return` on the same line as the `def` (or replace with `...` altogether, this doesn't really test anything). I had it like this in my example for a reason, I wanted to be sure there is no off-by-one error.

Reply:
Should I replace all the `return`s?

Reply:
Other returns are up to you, their positions are not really relevant